capture.js

```diff
@@ -0,0 +1,53 @@
+const capture = {
+
+  init() {
+    this.addListeners();
+    this.prevSite = null;
+    this.nextSite = null;
+  },
+
+  addListeners() {
+    // listen for each HTTP response
+    browser.webRequest.onResponseStarted.addListener(this.collectRequests, {urls: ['<all_urls>']}
+    );
+    // listen for when user visits a new site
+    browser.tabs.onUpdated.addListener(this.checkNewSite);
+  },
+
+  async collectRequests(response) {
+    const tab = await browser.tabs.get(response.tabId);
+    let tabUrl = new URL(tab.url);
+    tabUrl = tabUrl.hostname;
+    let targetUrl = new URL(response.url);
+    targetUrl = targetUrl.hostname;
+    let originUrl = new URL(response.originUrl);
+    originUrl = originUrl.hostname;
+
+    // capture third party requests
+    if (targetUrl !== originUrl) {
+      const thirdPartyData = {
+        document: tabUrl,
+        target: targetUrl,
+        origin: originUrl,
+        requestTime: response.timeStamp
+      };
+      console.log('storage.thirdPartyRequest:', tabUrl, thirdPartyData);
+    }
+  },
+
+  checkNewSite(tabId, changeInfo, tab) {
+    if (changeInfo.url) {
+      this.nextSite = changeInfo.url;
+    }
+    //favicon not available until tab status is complete; opening a new tab (about:blank) does not trigger a call to storage API
+    if ((this.nextSite !== this.prevSite) && (changeInfo.status === 'complete') && (this.nextSite !== null) && (this.nextSite !== 'about:blank')) {
```

Comment:
Code style nit, line length is likely wider than 80 chars these days but I would break by statement here:
```
 if (this.nextSite !== this.prevSite
     && changeInfo.status === 'complete'
     && this.nextSite !== null
     && this.nextSite !== 'about:blank') {
```